### PR TITLE
Drivers: hwinfo: add reset cause shell

### PR DESCRIPTION
**Pull request by alexanderwachter**
This PR adds a shell command to obtain, clear and show available reset causes

### CODE REVIEW

File: drivers/hwinfo/hwinfo_shell.c [hunk 1]
```diff
@@ -38,8 +38,149 @@ static int cmd_get_device_id(const struct shell *shell, size_t argc, char **argv
...
+		shell_print(shell, "No causes supporte");
+	}
+
+	return 0;
+}
+
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_reset_cause,
+	SHELL_CMD_ARG(show, NULL, "Show the persistent reset cause",
+		      cmd_show_reset_cause, 1, 0),
+	SHELL_CMD_ARG(clear, NULL, "Clear all persisten reset causes",
```
Comment (pdgendt) [inline drivers/hwinfo/hwinfo_shell.c:172]:
```suggestion
	SHELL_CMD_ARG(clear, NULL, "Clear all persistent reset causes",
```

File: drivers/hwinfo/hwinfo_shell.c [hunk 2]
```diff
@@ -38,8 +38,149 @@ static int cmd_get_device_id(const struct shell *shell, size_t argc, char **argv
...
+static void print_all_reset_causes(const struct shell *shell, uint32_t cause)
+{
+	for (uint32_t cause_mask = 1; cause_mask; cause_mask <<= 1) {
+		if ((cause & cause_mask) == 0) {
+			continue;
+		}
+
+		shell_print(shell, "- %s", cause_to_string(cause & cause_mask));
+	}
+}
```
Comment (pdgendt) [inline drivers/hwinfo/hwinfo_shell.c:94]: Why not just:
```suggestion
static void print_all_reset_causes(const struct shell *shell, uint32_t cause)
{
	for (uint32_t cause_mask = 1; cause_mask; cause_mask <<= 1) {
		if (cause & cause_mask) {
			shell_print(shell, "- %s", cause_to_string(cause & cause_mask));
		}
	}
}
```

**Review by pdgendt — APPROVED** [review]
Small nits, otherwise LGTM

File: drivers/hwinfo/hwinfo_shell.c (hunk 2 above)
Comment (alexanderwachter) [reply drivers/hwinfo/hwinfo_shell.c:94]: Whoops 😅
Much easier, yes!

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: drivers/hwinfo/hwinfo_shell.c [hunk 3]
```diff
@@ -19,27 +19,165 @@ static int cmd_get_device_id(const struct shell *shell, size_t argc, char **argv
...
+		return "clock";
+
+	default:
+		return "Unknown";
+	}
+}
+
+static void print_all_reset_causes(const struct shell *sh, uint32_t cause)
+{
+	for (uint32_t cause_mask = 1; cause_mask; cause_mask <<= 1) {
```
[the remark this answers is not in the document]
Comment (alexanderwachter) [reply drivers/hwinfo/hwinfo_shell.c:87]: A minimal scope is always preferable.

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by alexanderwachter** [comment]
@cfriedt ready to merge :-)

**Comment by cfriedt** [comment]
Cc @arvinf

**Pull request merged**